### ISSUE TITLE
All components callable via REST API?

### ISSUE THREAD

**Issue by dkoslicki** [RTXteam/RTX#619]
In our Milestones, we promised to make sure that ARAX_{Overlay, Expander, Filter} would be callable via REST API. Do we have this feature via the `ARAX_query.py` and tell them to use the `processing_actions` portion of the "API Standard Format 0.9.3"? Or would we rather stand up three different API endpoints that expose overlay/expand/filter?

**Comment by saramsey** [comment]
My $0.02:  for the demo, I'm fine with either. I don't feel that we are *required* to provide a unified endpoint at this time (though if that is feasible and useful, I think it is fine to have a single endpoint).

**Comment by edeutsch** [comment]
The plan is a unified endpoint. The old RTXQuery was a unified endpoint and ARAXQuery is modeled after that. In fact, the current standard supports the unified endpoint while multiple endpoints would be going outside the standard. I view it as a substantial benefit that all processing can happen sequentially with one call without have to manually pass it back and forth to different endpoints (which you can still do if you really wanted to).

In short, yes, we're already there, except for some minor plumbing work which I'll do when I return.

**Comment by dkoslicki** [comment]
Sounds good! I concur with having it nice to be able to do things sequentially rather than passing back and forth.

I'll leave this issue open just as a reminder for the minor plumbing work when you get back

**Comment by edeutsch** [comment]
This now works with the latest code in demo. If you start the swagger_server on your local dev machine, then you can run this test query which runs commands via the API and it should work:
b386ecafe7ad49f9c19ec8b266dcd865697be048
This has not been deployed on arax.rtx.ai yet, but I suppose that is the next step!

**Comment by dkoslicki** [comment]
Any plans for this @edeutsch since it was in the milestones? Or is the iPython notebooks "good enough" at this point (instead of dedicated REST API endpoints, which we technically promised)?

**Comment by edeutsch** [comment]
Did we promise dedicated REST API endpoints?
I don't quite understand the "iPython notebooks 'good enough'".. The iPython Notebooks are *demonstrating* the REST API.
You can call each step separately if you want, no problem. You just need to keep passing back and forth the Message. but that's not a problem.

I would argue all components are callable (individually or in series) via a REST API through a single endpoint.
If that seems less than optimal, we should discuss.
I would argue this is done.

**Comment by dkoslicki** [comment]
Yes, we did promise that. The milestones are:

1. Create a prototype of ARAX-Overlay , a software tool for overlaying weights (e.g.,
semantic distance, clinical co-occurrence frequencies, etc.) on a given input graph. The
prototype will be overlay at least two types of information on an input graph: co-
occurrence information from the Columbia Open Health Dataset (COHD) and semantic
distance in the biomedical literature (normalized Google distance computed using
PubMed abstracts). **ARAX-Overlay will be callable via a REST API**.
2. Create a prototype of ARAX-Expander that will expand an input graph based on edges
from the RTX knowledge graph (already created during the Translator feasibility
assessment phase), KG2. **ARAX-Expander will be callable via a REST API**.
3. Create a prototype of ARAX-Filter, which will enable refinement of an input graph based
on a filter string expressed in a domain-specific language (DSL). This will support refining
a graph based on a variety of optional criteria including edge type, knowledge source
type, user-specified thresholds, or property-value assertions. **ARAX-Filter will be callable
via a REST API**.

My question ways more: do we need __individual__ REST API's, or is the single unified API with DSL commands enough to mark this milestone as complete (as demonstrated in the iPython notebooks).

If you're fine with saying they are all callable via REST API (it just happens to be a single endpoint), then I am happy

**Comment by edeutsch** [comment]
Yes, I advocate that we can say they are they are all callable via a REST API through a single end point such that they can be chained together or not as desired.

I think we're there already.

**Comment by dkoslicki** [comment]
@edeutsch shall we close per #788?

**Comment by edeutsch** [comment]
I think so, yes.
